lib/memo_wise.rb

```diff
@@ -243,7 +238,14 @@ def #{method_name}(#{MemoWise::InternalAPI.args_str(method)})
       end
     end
 
-    unless target.singleton_class?
+    if target.singleton_class?
+      # This ensures that a memoized method defined on a parent class's
+      # singleton class can still be used by a child class.
+      def (MemoWise::InternalAPI.original_class_from_singleton(target)).inherited(subclass) # rubocop:disable Lint/NestedMethodDefinition
```

Comment:
One thing I don't fully understand—tests pass when I change this to just `def inherited(subclass)` (except the one test that checks which `public_instance_methods` are added, so behavior _is different_). @jemmaissroff and I dug in a little bit but couldn't figure out why this approach works. @ms-ati do you know?

Reply:
Could this be a bit more readable / accessible to readers as:

```suggestion
      parent_class = MemoWise::InternalAPI.original_class_from_singleton(target)
      def parent_class.inherited(subclass) # rubocop:disable Lint/NestedMethodDefinition
```